internal/controllers/reconciliation/controller.go

```diff
@@ -116,14 +116,15 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 		}
 	}
 
-	c.resourceClient.PatchStatusAsync(ctx, &req.Manifest,
-		func(rs *apiv1.ResourceState) bool {
-			return !((resource.Deleted() && !rs.Deleted) || !rs.Reconciled)
-		},
-		func(rs *apiv1.ResourceState) {
-			rs.Deleted = resource.Deleted()
-			rs.Reconciled = true
-		})
+	c.resourceClient.PatchStatusAsync(ctx, &req.Manifest, func(rs *apiv1.ResourceState) *apiv1.ResourceState {
```

Comment:
Did you move to this pattern to favor immutability? Or did something else in the code made this pattern not ergonomic?